Medium/solution.js

```diff
@@ -0,0 +1,65 @@
+class Solution {
+    /**
+     * Solução:
+     * 
+     *  Passo 1 - Crio um contador que amazenará a frequência de cada numero
+     *      em nums. E também crio um array de listas para armazenar numeros conforme
+     *      a frequência de aparição deles
+     * 
+     *  Passo 2 - Faço um loop que pecorre cada numero do array 'nums' e amazeno contador.
+     *      Também faço uma verificação se esse numero já está presente na contagem, caso
+     *      não exista será 0 o seu valor inicial
+     * 
+     *  Passo 3 - Faço outro loop que pecorrer o contador para cada numero ele adiciona 
+     *      ele adiciona o numero no array da frequencia criado no passo 1
+     * 
+     *  Passo 5 - Crio um array vazio que irá armazenar os elementos mais frequentes.
+     *  
+     *  Passo 6 - Crio mais um loop que pecorrer o array da frequencia da direita para
+     *      a esquerda(começando da frequencia mais alta)
+     * 
+     *  Passo 7 - Crio mais um loop que para cada lista do array da frequencia ele irá
+     *      inserir esses numeros no array criado no passo 5
+     * 
+     *  Passo 8 - Crio uma condição se o array do passo 5 atingir o tamanho igual do segundo
+     *      parametro passado na função, ele retorna o array.
+     * 
+     *  Tempo necessario: 2h
+     * 
+     *  Big O Espaço: O(n) no pior caso, pois o contador e frequencia também irão 
+     *                  depender da quantidade de numeros
+     *  Big O Tempo: O(n) no pior caso, pois vai depender da quantidade de numeros
```

Comment:
Acho que os valores estão errados aqui.

Link de referencia para atualizar para o valor certo:

- https://interviewing.io/questions/top-k-frequent-elements
- https://dev.to/lofiandcode/big-o-part-3-log-of-n-3n2h#:~:text=O(log%20N)%20is%20a,of%20O(log%20N).
- https://www.freecodecamp.org/news/big-o-notation-why-it-matters-and-why-it-doesnt-1674cfa8a23c/

Reply:
Eu dei uma olhada, Mas ainda acho que que tanto o espaço e tempo ainda estão com a complexidade **_O(n)_**.

Tempo:

- O primeiro loop percorre todos os elementos do array `nums`. Para cada número, ele atualiza o dicionário `count`, incrementando o valor do número de ocorrências desse elemento. Portanto, a complexidade desse loop é **_O(n)_**
`n` é o tamanho do array `nums`.
- O segundo loop o `count` tem no máximo `n` entradas (caso todos os elementos sejam únicos). No pior caso, percorrer todas as chaves de `count` levará **_O(n)_** e os métodos `push` e `parseInt`  são O(1).
- Resultando em uma complexidade de **_O(n)_**.

Espaço:

- `count` pode armazenar até `n` entradas gerando um espaço de _**O(n)**_.
- `freq` tem o tamanho de `n + 1`, mas cada posição contém um array que pode armazenar vários números. No pior caso, todos os elementos de `nums` podem ter frequências diferentes, o que significa que cada entrada de `freq` pode conter até `n` elementos no total. O que também resultaria **_O(n)_**.
- Resultando em uma complexidade de _**O(n)**_.

Reply:
Faz sentido 🙌 . Vlw pela explicação 👍 🚀